### DISCUSSION TITLE
Where are provider logs stored?

### DISCUSSION THREAD

**Discussion by atifsyedali** [wasmCloud/wasmCloud#3373] [Q&A] [answered]
I noticed that the provider logs don't appear in the default log output of `wash up`, unlike the component logs. I even tried to set the log level to TRACE. Where can I find the provider logs?

**Comment by atifsyedali** [comment]
Also tried to use the wasi:logging WIT interface in a provider and with that, it just hangs.

**Comment by atifsyedali** [comment]
Looks like provider logs show up in the wash logs with a golang provider but not with a rust provider.

**Comment by vados-cosmonic** [comment, marked as answer]
Hey @atifsyedali so by default the [stdout for providers are *not* forwarded](https://github.com/wasmCloud/wasmCloud/blob/main/crates/host/src/wasmbus/mod.rs#L2312) -- what I think is happening with the Golang provider is that the logging calls are converted under the covers. 

That said the normal behavior is that you *should* be able to use the `wasi:logging` interfaces to log to the host from a provider. I think you might have run into some issues due to a sudden upstream `wasi:logging` change -- due to using the contracts [on `wasi-logging` `main` branch](https://github.com/WebAssembly/wasi-logging/commits/main/) changing on us.

The fix that we merged last week is [here](https://github.com/wasmCloud/wasmCloud/commit/5095c22d266331cecdeacc0d29b0a4d08f5ab258) (importantly the `wasi:logging/logging@0.1.0-draft`), A [release is already cut](https://github.com/wasmCloud/wasmCloud/commit/f5e3e86fc2e9f0b94d59d84f7ce8b6b37cdf5a92) (the very next commit), and the examples updated.

Soon we'll have a new version of `wash-cli` out that uses this new host by default, and you should be able to use `wasi:logging`. Apologies for the delay and thank you for your patience here!

[EDIT] Here's an issue that is directly related:

https://github.com/wasmCloud/wasmCloud/issues/3287#event-14761164692
